draft/0002-management-api-scopes.md

```diff
@@ -0,0 +1,127 @@
+---
+Start date: 2023-09-27
+Author(s): gao
+Revision: 1
+---
+
+# Management API scopes
+
+## Overview
+
+In Logto, machine-to-machine applications can utilize the Management API to programmatically access Logto resources. This proposal suggests the addition of scopes to the Management API to enhance access control.
+
+## Motivation
+
+Currently, the Management API offers only the `all` scope, granting complete access. This isn't ideal from both security and usability perspectives.
+
+For instance, a user might want to provide a third-party application with read-only access to logs without permitting log creation or deletion.
+
+The introduction of scope sets can also facilitate Logto Cloud's collaboration feature, which lays the groundwork for various roles in the future.
+
+## Detailed explanation
+
+### Scope composition
+
+Each scope will adhere to the naming convention of `action:resource`. This division is easily comprehensible to users and suits RESTful API design.
+
+We define the following actions as the initial set:
+
+- `read`: for reading resources
+- `write`: for creating or updating resources
+- `delete`: for deleting resources
+
+Here's a table listing resources and their associated actions:
+
+| Resource      | Read | Write | Delete | Description                                                                          |
+| ------------- | ---- | ----- | ------ | ------------------------------------------------------------------------------------ |
+| dashboard     | Y    | N     | N      | Statistics for new, active, and total users.                                         |
+| apps          | Y    | Y     | Y      | All apps in the tenant.                                                              |
+| resources     | Y    | Y     | Y      | All API resources and their permissions in the tenant except for the Management API. |
```

Comment:
try to confirm: some actions such as managing corresponding scopes of the resource will share the `write:resource` permission.

Reply:
Btw, we currently do not prevent API from managing "Logto management API" of the tenant, not sure whether we should align this (we do not have a delete and add permission button for "Logto management API").